src/shared_modules/utils/tests/CMakeLists.txt

```diff
@@ -2,6 +2,28 @@ cmake_minimum_required(VERSION 3.12.4)
 
 project(utils_unit_test)
 
+include(CheckCXXSourceCompiles)
+
+# Check if std::to_chars supports floating-point numbers
+check_cxx_source_compiles("
+    #include <charconv>
+    #include <system_error>
+    int main() {
+        char buffer[32];
+        float f = 3.14f;
+        auto [ptr, ec] = std::to_chars(buffer, buffer + sizeof(buffer), f);
+        return ec == std::errc() ? 0 : 1;
+    }
+" HAS_TO_CHARS_FLOAT)
+
+if(HAS_TO_CHARS_FLOAT)
+  add_definitions(-DHAS_TO_CHARS_FLOAT=true)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
I think the standard is already defined at `src/shared_modules/utils/CMakeLists.txt`.
Also, isn't enough with this ?
```
if (${TARGET} STREQUAL "server")
```

Reply:
When compiling with ASAN on Ubuntu 20.04 with the server target, the compilation fails. After researching the issue, I followed the approach defined [here](https://github.com/wazuh/wazuh/pull/17982/commits/4c84b17b87539a43537f78408de23792aa7e7437#r1287909903). Since we share utilities with the agent, perhaps we should conditionally compile this test only if the C++17 standard is available.

Please let me know if this approach aligns with your expectations or if you'd prefer we consider an alternative solution :)

Reply:
This feature requires a C++17 standard

Reply:
I've tested a package from this PR, and I didn't find any problem to process the elements to index.